### ISSUE TITLE
[BUG][JAVA] Invalid code generated: ApiClient is referred but not present

### ISSUE THREAD

**Issue by sephiroth-j** [OpenAPITools/openapi-generator#19142]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
After upgrading from v7.6.0 to v7.7.0, we have noticed that the generated code is invalid.

- The generated model class has an import of a class `ApiClient` which is not generated.
- The generated model class has a function `toUrlQueryString(String)` that uses a static function `valueToString` from the class `ApiClient` which is not generated.
The old code was
```java
joiner.add(String.format("%sfoo%s=%s", prefix, suffix, URLEncoder.encode(String.valueOf(getFoo()), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
```
The new code is
```java
joiner.add(String.format("%sfoo%s=%s", prefix, suffix, URLEncoder.encode(ApiClient.valueToString(getFoo()), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
```

##### openapi-generator version

7.7.0

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.3
info:
  title: Example API v1
  description: A reference API
  contact: {}
  version: 2.0.0
paths:
  /demo:
    get:
      responses:
        '200':
          description: Test
          content:
            'application/json':
              schema:
                type: object
                properties:
                  foo:
                    type: string
```

##### Generation Details
language: java
library: native

```xml
				<groupId>org.openapitools</groupId>
				<artifactId>openapi-generator-maven-plugin</artifactId>
				<version>7.7.0</version>
				<configuration>
					<generatorName>java</generatorName>
					<skipValidateSpec>false</skipValidateSpec>
					<generateApis>false</generateApis>
					<generateSupportingFiles>false</generateSupportingFiles>
					<generateModelTests>false</generateModelTests>
					<generateModelDocumentation>false</generateModelDocumentation>
					<generateApiTests>false</generateApiTests>
					<generateApiDocumentation>false</generateApiDocumentation>
					<output>${project.build.directory}/generated-sources</output>
					<configOptions>
						<sourceFolder>openapi</sourceFolder>
						<useJakartaEe>true</useJakartaEe>
						<withXml>true</withXml>
						<serializationLibrary>jackson</serializationLibrary>
						<library>native</library>
						<dateLibrary>java8</dateLibrary>
						<disallowAdditionalPropertiesIfNotPresent>false</disallowAdditionalPropertiesIfNotPresent>
					</configOptions>
				</configuration>
				<executions>
					<execution>
						<id>demo</id>
						<goals>
							<goal>generate</goal>
						</goals>
						<phase>generate-sources</phase>
						<configuration>
							<inputSpec>src/main/openapi/demo.yaml</inputSpec>
							<configOptions>
								<modelPackage>de.demo</modelPackage>
							</configOptions>
						</configuration>
					</execution>
				</executions>
```

##### Steps to reproduce

1. save the file above as `src/main/openapi/demo.yaml`
2. add the above plugin configuration to a maven pom
3. run `mvn generate-sources`
4. check `target/generated-sources/openapi/de/demo/DemoGet200Response.java`

**Comment by wing328** [comment]
@sephiroth-j thanks for reporting the issue

for your use cases, you only need the model files, right?

can you also generate the ApiClient.java as well (generateSupportingFiles set to true)? or you want to avoid extra dependencies in your model files?

**Comment by welandaz** [comment]
> @sephiroth-j thanks for reporting the issue
> 
> for your use cases, you only need the model files, right?
> 
> can you also generate the ApiClient.java as well (generateSupportingFiles set to true)? or you want to avoid extra dependencies in your model files?

Hi @wing328 !

For our case, we do generate the `ApiClient.java` as well by default.
P.S. As a side note, we use a Gradle plugin and that doesn't seem to have the `generateSupportingFiles` option you mentioned.

**Comment by wing328** [comment]
can you push the auto-generated java client to a github repo so that I can take a further look?

**Comment by welandaz** [comment]
> can you push the auto-generated java client to a github repo so that I can take a further look?

In our case, the class is generated and the `valueToString` function is present in it.
However, we do get an error that the package where the `ApiClient.java` resides does not exist.

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/pull/18743/files#diff-2eefc7a98c8e6aff3fa4589d8f2c364e7214c35e98806786e8dae207c42352dfR5

did you set the invokerPackage?

**Comment by welandaz** [comment]
> https://github.com/OpenAPITools/openapi-generator/pull/18743/files#diff-2eefc7a98c8e6aff3fa4589d8f2c364e7214c35e98806786e8dae207c42352dfR5
> 
> did you set the invokerPackage?

yes, we have that configured from the beginning

**Comment by wing328** [comment]
can you push the auto-generated java client to a github repo so that I can take a further look?

ideally with the spec and config options so that i can repeat the issue

**Comment by sephiroth-j** [comment]
> @sephiroth-j thanks for reporting the issue
> 
> for your use cases, you only need the model files, right?
> 
> can you also generate the ApiClient.java as well (generateSupportingFiles set to true)? or you want to avoid extra dependencies in your model files?

Yes exactly, we are only interested in the model files. Other files, especially the ApiClient, are not needed/wanted.

**Comment by wing328** [comment]
@sephiroth-j 

for your use case, you can use the workaround as you suggested since `toUrlQueryString` is only needed for API operations (e.g. making HTTP calls with query strings), right?

> Set `supportUrlQuery` to `false` when `toUrlQueryString` is not needed.

**Comment by sephiroth-j** [comment]
Yes

**Comment by madorb** [comment]
Running into this as well due to the indicated regression.

**Comment by welandaz** [comment]
Just a note from our side: after having a deeper look, the root cause of the issue was related to our build setup. It came up because the model classes now depend on client one (e.g. `ApiClient`) which was not the case before, so we needed to include it explicitly for compilation.

**Comment by madorb** [comment]
@welandaz right, but when generating models they _shouldn't_ rely on client classes. I'd suggest the problem isn't with your build, but with the generator emitting model classes that depend on `ApiClient`. No matter what is included in a specific generator run - i say that just conceptually having `Model` classes depend on an `ApiClient` class is a code smell.

**Comment by JenniferDwinall** [comment]
I am setting `supportUrlQuery` to `false` but it does not work when using the `apache-httpclient` library:

```
    configOptions.set([
            library: "apache-httpclient",
            supportUrlQuery: "false"
    ])
```

```
error: cannot find symbol
          joiner.add(getAggregatedTotals().get(_key).toUrlQueryString(String.format("%saggregatedTotals%s%s", prefix, suffix,
                                                    ^
  symbol:   method toUrlQueryString(java.lang.String)
  location: class java.math.BigDecimal

```

**Comment by JenniferDwinall** [comment]
When using the apache-httpclient (java 8), toUrlQueryString workaround does not work. Confirmed this happens since 6.3.0

```
implementation "org.openapitools:openapi-generator-gradle-plugin:6.6.0"
```

```
    configOptions.set([
            library: "apache-httpclient",
            supportUrlQuery: "false"
    ])
```

```
error: cannot find symbol
          joiner.add(getAggregatedTotals().get(_key).toUrlQueryString(String.format("%saggregatedTotals%s%s", prefix, suffix,
                                                    ^
  symbol:   method toUrlQueryString(java.lang.String)
  location: class java.math.BigDecimal
```

**Comment by doctau** [comment]
I am running into this as well. I only want the model classes and nothing else. I have been using the following for a long tie with the maven plugin, and after upgrading it does not compile because the model classes refer to the non-existing class
```
<generateApis>false</generateApis>
<generateSupportingFiles>false</generateSupportingFiles>
<generateModelTests>false</generateModelTests>
<generateModelDocumentation>false</generateModelDocumentation>
```

Adding `<supportUrlQuery>false</supportUrlQuery>` does solve the problem, but unless I found this GH issues I would never have known to add it
